### DISCUSSION TITLE
Last (debian testing) update with kermel 6.10.6 won't boot

### DISCUSSION THREAD

**Discussion by Spaceboy60** [SpiralLinux/SpiralLinux-project#478] [Q&A]
Just get a black screen, have to use old kernel to boot.   Plus snapshots gone from grub menu.  Also, now when i try run snapper rollback i get error:  

Cannot detect ambit since default subvolume is unknown.  This can happen if the system was not setup for rollback.  The ambit can be specified using the --ambit option.

**Comment by geckolinux** [comment]
Hi there, `snapper rollback` is not to be used in SpiralLinux. Please follow these instructions:
https://github.com/SpiralLinux/SpiralLinux-project/wiki#working-with-btrfs-snapshots-and-rollbacks

Also, if the problem is just with a kernel update then it's best to simply select a different working kernel in GRUB and optionally uninstall the buggy kernel, no need to rollback the system.

> snapshots gone from grub menu

Was this before or after you tried the `snapper rollback` command?

**Reply by Spaceboy60** [reply]
> Was this before or after you tried the `snapper rollback` command?

Before.  

Also according to the above instructions i should use: ` sudo snapper-rollback X where X is the number of the snapshot to rollback to.  `   However when i run the command i get the error:  `root argument can only be used together with no-dbus`

In terms of  kernel, i've removed the current kernel, so that's not an issue.

**Reply by geckolinux** [reply]
Sorry it didn't work. I imagine that the `snapper rollback` command broke something. I just tried an upgrade to Testing and then a rollback and it worked:

![Screenshot from 2024-09-06 01-01-49](https://github.com/user-attachments/assets/4cd174e0-d854-402c-bc86-a25e0ce15cce)

![Screenshot from 2024-09-06 01-00-05](https://github.com/user-attachments/assets/78f8c586-11f3-4f2c-9069-c5d1c96e8951)

![Screenshot from 2024-09-06 01-03-21](https://github.com/user-attachments/assets/97a79cb3-cc25-4e9f-8f3d-d8c4cc0c29ff)

![Screenshot from 2024-09-06 01-04-54](https://github.com/user-attachments/assets/55027b17-69d5-4012-bd8a-438db049454c)

**Reply by Spaceboy60** [reply]
> Sorry it didn't work. I imagine that the `snapper rollback` command broke something. I just tried an upgrade to Testing and then a rollback and it worked:
> 
> ![Screenshot from 2024-09-06 01-01-49](https://private-user-images.githubusercontent.com/2692138/365052199-4cd174e0-d854-402c-bc86-a25e0ce15cce.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MTM1ODcsIm5iZiI6MTcyNTYxMzI4NywicGF0aCI6Ii8yNjkyMTM4LzM2NTA1MjE5OS00Y2QxNzRlMC1kODU0LTQwMmMtYmM4Ni1hMjVlMGNlMTVjY2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDkwMTI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzY2OTc2MjY5Y2UwZDFkYzQ5ZGJlNGQ5NjNkZDc1MDkzNWQzNGNkNTVkM2ViNThjOWMxNGI2MTRjZDdjM2Y0MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.tNgIv6362qZb-7y4rA_NDqxdZAy0TjyD9hgSIzoYR64)
> 
> ![Screenshot from 2024-09-06 01-00-05](https://private-user-images.githubusercontent.com/2692138/365052246-78f8c586-11f3-4f2c-9069-c5d1c96e8951.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MTM1ODcsIm5iZiI6MTcyNTYxMzI4NywicGF0aCI6Ii8yNjkyMTM4LzM2NTA1MjI0Ni03OGY4YzU4Ni0xMWYzLTRmMmMtOTA2OS1jNWQxYzk2ZTg5NTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDkwMTI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTM4MWU5ZDc0NGU5MjczZmZjOGRmMzVjMjVhY2M0MWMwMzBlZGNlOTQ5NTJlZDhiZjU5Yzc4NGQzNTE2NmE1ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.OKBnLNwSFMThDZPwsbcYBUO-jCA7GIHAX6Jgy6i226Y)
> 
> ![Screenshot from 2024-09-06 01-03-21](https://private-user-images.githubusercontent.com/2692138/365052383-97a79cb3-cc25-4e9f-8f3d-d8c4cc0c29ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MTM1ODcsIm5iZiI6MTcyNTYxMzI4NywicGF0aCI6Ii8yNjkyMTM4LzM2NTA1MjM4My05N2E3OWNiMy1jYzI1LTRlOWYtOGYzZC1kOGM0Y2MwYzI5ZmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDkwMTI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzhiMjc5ZmQxY2JjMTM5OTVmZTZmN2Q0MWRiMDBmYjc4OTlkNWFkNjNiOGIwNTQyOTQyZjk1YjE2MzEzNDEzNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.WWJZs4WIj37pWYVaL3NhewtMXXQ7wtaQ_rkA39owXAo)
> 
> ![Screenshot from 2024-09-06 01-04-54](https://private-user-images.githubusercontent.com/2692138/365052406-55027b17-69d5-4012-bd8a-438db049454c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MTM1ODcsIm5iZiI6MTcyNTYxMzI4NywicGF0aCI6Ii8yNjkyMTM4LzM2NTA1MjQwNi01NTAyN2IxNy02OWQ1LTQwMTItYmQ4YS00MzhkYjA0OTQ1NGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDkwMTI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTM2OTU5MTA0MjIwMzA3MzExY2RlM2JhMDBkOTMwODJhZjQzYTNlZGQ2OWI2MGE3MWE1YmNhZDZkODAwYjQ2ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ZvM5W5I5m8ReldGYBopzItAZ9s-vC9F3fKpYdO_A8kI)

And you did a full upgrade? 

Has it got anything to do with having a separate home partition? Any other ideas why these two issues may have occurred? 

I used apt update & & apt upgrade.

**Reply by Spaceboy60** [reply]
> I imagine that the `snapper rollback` command broke something

I doubt it, It was broken before that as no snapper option in grub menu and unable to boot after upgrade

**Reply by geckolinux** [reply]
> And you did a full upgrade?

Yep, I used this method:
https://github.com/SpiralLinux/SpiralLinux-project/wiki#switching-from-debian-stable-to-the-testing-or-unstable-branch

> Has it got anything to do with having a separate home partition? Any other ideas why these two issues may have occurred?

Not sure what might have happened, sorry I can't be more helpful. The separate `/home` partition won't make any difference, because even on a single `/` partition installation `/home` is on a separate Btrfs subvolume.

At this point the easiest way to recover the rollback functionality for your system would be to reinstall. Especially since you have a separate `/home` partition it should be quite painless (but still backup your data first). Just keep in mind that breakage in Debian Testing and Sid is to be expected.
https://github.com/SpiralLinux/SpiralLinux-project/wiki#switching-from-debian-stable-to-the-testing-or-unstable-branch
> **Important:** Debian's Testing and Unstable branches may work very well for some users' needs, but they do not claim to be stable releases and should only be used with extra care by experienced users.

**Comment by Spaceboy60** [comment]
I've reinstalled the budgie version, done the update again via synaptic, once again black screen upon reboot.  Last time i selected not replace lightdm and lightdm config, this time i selected to replace them both, same result: black screen with new kernel. And once again: `sudo snapper-rollback X where X is the number of the snapshot to rollback to` brings `root argument can only be used together with no-dbus` error.

**Reply by geckolinux** [reply]
Sorry I just saw your latest update. There are two separate issues here: 
1) Debian "Testing" as the name says is not meant to be a stable release. It's very common for different kernels in Testing to fix or break something depending on your specific hardware. When this happens, instead of doing a rollback it's best to select a different kernel in GRUB and/or uninstall the broken kernel.
2) I'm not sure why `snapper-rollback` isn't working for you, I'll look into this later.

**Reply by geckolinux** [reply]
Is your installation on a legacy BIOS or UEFI system?

**Reply by geckolinux** [reply]
I just did a fresh installation from `SpiralLinux_Budgie_12.231120_x86-64.iso` on a UEFI system with a separate `/home` partition, and upgraded it to Testing as described earlier.

At this point in time it looks like Budgie is broken in Testing, because it removes some core `budgie-*` packages and others are not installable. Upon reboot into the upgraded Testing system there was just an X session graphical terminal with no Budgie session available in LightDM.

So I rebooted into snapshot `#1` and performed a successful rollback:
![Screenshot from 2024-09-06 09-54-59](https://github.com/user-attachments/assets/514d31b7-9876-45db-afd7-b2c5f42d2ccb)

![Screenshot from 2024-09-06 09-56-42](https://github.com/user-attachments/assets/5cc9ca90-ab1a-4401-8515-60955139d042)

So if you could, please give a detailed list of the steps that you're taking when you reinstall (including the partition layout) as well as screenshots of the rollback attempts. Or if you prefer I would recommend you reinstall and stay on Debian 12 Stable for now, as Budgie is broken in Testing.

**Reply by Spaceboy60** [reply]
> Is your installation on a legacy BIOS or UEFI system?

Legscy

**Comment by Spaceboy60** [comment]
Ok, will do when I get back from hospital.

Would the fact the /home partition is ext4 make a difference?  / is btrfs.

And what may be causing the latest kernel to not book (black screen)?

Also if I select a snapshot in grub menu how do I make the rollback
permanent since you say you can't just do snapper rollback followed by a
reboot?

**Reply by geckolinux** [reply]
Hi there,

> Would the fact the /home partition is ext4 make a difference?

No, `snapper-rollback` only handles the Btrfs root partition.

> And what may be causing the latest kernel to not book (black screen)?

There's a million changes in every new kernel version, and in Debian Testing they're not guaranteed to work and frequently don't work.

> Also if I select a snapshot in grub menu how do I make the rollback permanent since you say you can't just do snapper rollback followed by a reboot?

Notice that `snapper` [space] `rollback` is not the same as `snapper-rollback` . They are completely different utilities. The instructions given for `snapper-rollback` ***do*** perform a permanent rollback after you reboot.

**Comment by Spaceboy60** [comment]
Just wanted to check/clarify couple of things :

sudo snapper-rollback x

1.There are no spaces after snapper, i.e snapper-rollback and not snapper
- rollback.

2. To get the rollback number I use sudo snapper list??

3. If I've booted from a read only snapshot I still need to specify a
snapshot number, i.e I can't just use sudo snapper-rollback

**Reply by geckolinux** [reply]
Hi there,

1. `sudo snapper-rollback X`
2. The number is visible when you select the snapshot to be booted in GRUB, and/or you can use **Snapper GUI** [(Instructions)](https://github.com/SpiralLinux/SpiralLinux-project/wiki#working-with-btrfs-snapshots-and-rollbacks)
3. Correct.

**Comment by Spaceboy60** [comment]
Thanks, not sure why but just did a sudo apt update & & sudo apt upgrade
and I'm presented with a black screen upon reboot.

**Reply by geckolinux** [reply]
Hi there, are you on Debian Testing or Stable?

**Comment by Spaceboy60** [comment]
Stable branch,  boots on old kernel still

**Reply by marwenius** [reply]
Perhaps problems with the graphics card drivers as discussed here:
https://github.com/orgs/SpiralLinux/discussions/448#discussioncomment-10467356

**Reply by geckolinux** [reply]
The easiest solution would probably be to just uninstall the broken kernel. If the Debian Stable 6.1.0 kernel works you could even remove the Backports kernel APT preference that SpiralLinux uses by default.

**Comment by Spaceboy60** [comment]
Yes that's what I ended up doing I'm using grub customizer to point to the
old kernel. I'll pay more attention to the kernel update version when I'm
updating next time. Or maybe freeze the kernel packages before updating.
